### PR TITLE
feat(desktop): v1 桌面前端整套移植、MCP 导入优化及启动韧性增强

### PR DESCRIPTION
**Pull request by SivanCola**
## 概要

本次 PR 包含两大块工作：v1 桌面前端整套移植及深度适配，以及 MCP 导入/配置/启动韧性增强。

### 1. v1 桌面前端整套移植

- 新增 `desktop/frontend/src/v1/`，共 55 个文件，涵盖 v1 UI 组件、样式、i18n、测试等。
- `desktop/frontend/src/App.tsx` 大幅改成 v1 风格 shell，接入 v1 的 Sidebar / Composer / StatusBar / SettingsModal / ContextPanel。
- `desktop/frontend/src/main.tsx` 引入 v1/styles.css 和 v1-compat.css。
- `desktop/frontend/src/v1-compat.css` 是这轮视觉兼容补丁，处理 Wails 拖拽、样式泄漏、会话行、按钮等问题。
- `desktop/frontend/src/styles.css` 也有大量 UI/主题/附件预览样式变更。

### 2. Wails 桌面桥接能力增强

- `desktop/app.go` 增加前端需要的桥接 API：tab 管理、MCP 配置、文件选择、打开路径、编辑器打开、图片附件读取等。
- `desktop/main.go` 改了 macOS titlebar 模式、菜单，并接入原生三色窗口按钮位置调整。
- 新增 `desktop/traffic_lights_darwin.go` / `traffic_lights_other.go`，用于调整原生 macOS 三色按钮位置。
- 新增 `desktop/install.go` / `install_test.go`，提供安装 `/usr/local/bin/reasonix` 命令的能力。

### 3. 图片粘贴 / 附件能力

- 前端新增 `ImagePreview.tsx`，`Composer.tsx`、`Message.tsx` 支持图片粘贴、附件缩略图和预览。
- 后端新增 `internal/control/attachments.go` / test，支持保存剪贴板图片、data URL、图片文件，并生成 `.reasonix/attachments/...` 引用。
- CLI 的 `chat_tui.go` 也增加了图片粘贴、`/paste-image`、附件 tray、发送时隐藏真实路径但 provider 侧保留引用。

### 4. MCP 导入 / 配置 / 状态管理

- `/mcp import` 改为交互式 cc-switch 导入选择器，而非无差别导入所有服务器。
- 导入的 MCP 服务器可标记 `auto_start = false`，保持配置但不随聊天启动。
- `/mcp connect <name>` 支持在运行中手动连接 MCP 服务器。
- MCP 启动韧性：失败的服务器被记录并汇总，而非中断聊天启动。
- Stdio MCP stderr 被捕获并汇总，不再直接写入 TUI。
- MCP 工具和提示名称规范化为 provider 安全的标识符（a-zA-Z0-9_-）。
- cc-switch MCP 导入兼容不同 sqlite 字段。
- MCP transport / timeout / auto_start 配置完善。
- 新增 add/update/retry/remove MCP server 的控制层能力。

### 5. 全新的设置页面

- 新增 `desktop/frontend/src/v1/ui/settings.tsx`，完整重构设置页面的 UI 架构。
- 精简原有冗余设置项，保留核心配置路径。
- 增加必要的新设置项，确保桌面端具备完整的功能控制能力。
- 预埋多模型商的配置结构，为后续接入多种 AI 提供商做好准备。

## 设计要点

基于 v1 版本的桌面做了深度适配和精简，保证了绝大多数功能的可用性，后期基于此进行后续开发，可以很快上线正式版的 v2。
【ps】
合并本分支后， 1，v2版桌面app就可以有一定的使用性；2，由于本次对 mcp 的扩展和图片粘贴的增强，可以实现基于 mcp 的图像识别，为 deepseek 模型增加‘一双眼睛’;3,基于此，可以快速迭代 v2 桌面版的新功能。

## 验证

- `go test ./...`
- `cd desktop && go test ./...`
- `make build`

### CODE REVIEW

**Comment by SivanCola** [comment]
@codex review

File: internal/config/ccswitch.go [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+// older installs kept them in config.json(.migrated/.bak), so we support both.
+func LoadCCSwitchMCP() ([]PluginEntry, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return nil, fmt.Errorf("cc-switch import: resolve home: %w", err)
+	}
+	root := filepath.Join(home, ccSwitchDir)
+	if entries, err := loadCCSwitchMCPDB(filepath.Join(root, "cc-switch.db")); err == nil && len(entries) > 0 {
+		return entries, nil
+	} else if err != nil && !os.IsNotExist(err) {
```
[the remark this answers is not in the document]
Comment (SivanCola) [reply internal/config/ccswitch.go]: Fixed in d8d5f485. The cc-switch SQLite database is now authoritative whenever it exists and can be read, even when the Codex-enabled query returns zero rows. In that case we return an empty candidate list instead of falling back to stale legacy JSON backups. I also added a regression test with an empty SQLite DB plus a stale config.json.migrated backup to lock this behavior.

Verification:
- go test ./internal/config
- go test ./...
- cd desktop && go test ./...

**Comment by SivanCola** [comment]
Updated the PR to include latest `origin/main-v2` and fixed the CI gofmt failure in `internal/agent/storm_test.go` with commit d52af046.

Local verification:
- root gofmt gate: `unformatted=$(gofmt -l . | grep -v '^desktop/' || true); if [ -n "$unformatted" ]; then echo "$unformatted"; exit 1; fi`
- `go test ./...`

Note: `cd desktop && go test ./...` currently requires the generated `desktop/frontend/dist` asset tree in this worktree and failed before running the desktop package with `pattern all:frontend/dist: no matching files found`; this is separate from the root CI gofmt failure reported in the linked job.

**Comment by SivanCola** [comment]
@codex review

**Comment by SivanCola** [comment]
Addressed the Codex review feedback in `6c6d9b42`:

- `@` file references now resolve relative to the controller workspace root, so desktop tabs read files from the selected workspace instead of the launch directory.
- MCP config read/write paths now use the controller workspace root through `LoadAt`/`SaveAt`, so add/edit/remove/import operations persist to the active workspace.
- Relative `[sandbox] workspace_root` values are now anchored to the selected workspace in `WriteRootsAt`, matching `allow_write` handling.

Added regression tests for all three cases. Verified locally with `gofmt`, `go vet ./...`, `go build ./...`, and `go test ./...`; CI run 26723489997 is passing on Ubuntu and macOS.

中文：已修复 Codex review 的三类 workspace/cwd 问题，并补了对应回归测试。本地和远端 CI 均通过。

**Comment by esengine** [comment]
Thanks for the substantial work here, @SivanCola — closing this one on architectural grounds, and want to explain why so the effort isn't lost.

v2 is a deliberate from-scratch rewrite (Go kernel + a natively rebuilt **Wails** desktop), not a port of the v1 frontend. This PR is ~33k added lines, and ~27k of those (55 files) are a verbatim copy of the v1 desktop frontend under `desktop/frontend/src/v1/` — the v1 i18n set, theme, notifications, CommandPalette, Markdown, plus `tauri-compat.ts` and `v1-compat.css` shims. v1's desktop was Tauri-based; pulling that surface and its compat layers into the Wails app runs against the rewrite's whole premise and would commit us to maintaining two desktop frontends.

That said, several pieces in here are genuinely valuable and in-scope — we'd happily take them as small, focused PRs:
- the MCP import optimization
- the startup-resilience improvements (`internal/boot`)
- the targeted component improvements (Composer/Message/bridge)

The autoscroll fix from this branch already landed standalone as #2487. Closing in favor of splitting out those targeted improvements rather than the wholesale v1 import. Really appreciate the contribution — thank you.
